libnavigation-copilot/src/main/java/com/mapbox/navigation/copilot/MapboxCopilotImpl.kt

```diff
@@ -48,9 +53,12 @@ import java.util.Locale
 @OptIn(ExperimentalPreviewMapboxNavigationAPI::class)
 internal class MapboxCopilotImpl(
     private val mapboxNavigation: MapboxNavigation,
+    private val computationDispatcher: CoroutineDispatcher = ThreadController.DefaultDispatcher
 ) {
 
     private val mainJobController by lazy { InternalJobControlFactory.createMainScopeJobControl() }
+    private var recordingSessionScope: CoroutineScope =
+        InternalJobControlFactory.createMainScopeJobControl().scope
```

Comment:
Scopes created via `InternalJobControlFactory` are not cancelled on destroy.
So it looks like you never cancel it (and when you cancel, you recreate). Should we properly cancel it on destroy?

Reply:
Points from out discussion in call: we have found obvious solution of how to organise the scope, all considered options have some disadvantages. It seems that we need to reconsider the proper way later. 
@Guardiola31337 , what are your thoughts on that?

Reply:
What about creating a job and canceling the job instead of canceling the scope? Have you considered that? That way we don't need to worry about leaving an orphan scope - job is created always and canceled always, no need to recreate when stopping, you create it when it's launched.

Reply:
done